lib/benches/parser.rs

```diff
@@ -0,0 +1,49 @@
+#![feature(test)]
+
+extern crate test;
+use test::{black_box, Bencher};
+
+macro_rules! parser {
+	($name: ident, $parser: path, $text: expr) => {
+		#[bench]
+		fn $name(b: &mut Bencher) {
+			let text = $text;
+			b.iter(|| {
+				black_box($parser(black_box(text))).unwrap();
+			});
+		}
+	};
+}
+
+parser!(select_simple, surrealdb::sql::parse, "SELECT * FROM person;");
+parser!(
+	select_complex,
+	surrealdb::sql::parse,
+	"SELECT name, age, country FROM person WHERE hair = \"brown\" AND is_vegetarian;"
+);
+parser!(transaction, surrealdb::sql::parse, "BEGIN TRANSACTION; UPDATE person:finn SET squirrels = \"yes\"; SELECT * FROM person; COMMIT TRANSACTION;");
+parser!(datetime, surrealdb::sql::parse, "RETURN \"2022-07-03T07:18:52.841147+02:00\";");
+parser!(duration, surrealdb::sql::parse, "RETURN [100w, 5d, 20m, 2s];");
+parser!(casting_deep, surrealdb::sql::parse, "RETURN <float><float><float><float><float>1.0;");
+parser!(
+	json_geo,
+	surrealdb::sql::parse,
+	"RETURN { type: \"Point\", coordinates: [-0.118092, 51.509865] };"
+);
+parser!(json_number, surrealdb::sql::json, "1.2345");
+parser!(
+	json_small_object,
+	surrealdb::sql::json,
+	"{\"key\": true, \"number\": 42.0, \"value\": null}"
```

Comment:
How about using a raw string here so we don't have to use the escape character?

Reply:
I switched them to single quotes (like everything else) since I think having a quote character is more representative of real JSON.

Reply:
JSON doesn't support single quotes though. Does `surrealdb::sql::json` take care of that? I think it does but I'm not sure.

Reply:
Never mind, I've just checked. It does!

Reply:
Yeah, surrealdb's parser is lenient. Note that, if it wasn't, the `unwrap()` in each benchmark would immediately surface the parsing error rather than benchmarking garbage.